### PR TITLE
apps: use api calls when creating dc in extended tests

### PR DESCRIPTION
**Pull request by mfojtik**
This will make all deployments extended tests use API calls when creating the fixtures instead of using the CLI (`create -f`). Also makes sure the failure traps we have use the correct deployment config names.

/cc @tnozicka 

(this might also reveal how many broken failure traps we have ;-)

### CODE REVIEW

File: test/extended/deployments/util.go [hunk 1]
```diff
@@ -625,24 +614,17 @@ func readDCFixture(path string) (*appsapi.DeploymentConfig, error) {
...
-	return dc, err
-}
-
-func readDCFixtureOrDie(path string) *appsapi.DeploymentConfig {
-	data, err := readDCFixture(path)
-	if err != nil {
-		panic(err)
-	}
-	return data
+	appsScheme := runtime.NewScheme()
```
Comment (mfojtik) [inline test/extended/deployments/util.go]: @deads2k @tnozicka double check? not sure why the tests were failing

Comment (deads2k) [reply test/extended/deployments/util.go]: > @deads2k @tnozicka double check? not sure why the tests were failing

what error were you getting?

Comment (mfojtik) [reply test/extended/deployments/util.go]:
```
    Expected error:
        <*runtime.notRegisteredErr | 0xc42194cfa0>: {
            gvk: {
                Group: "",
                Version: "v1",
                Kind: "DeploymentConfig",
            },
            target: nil,
            t: nil,
        }
        no kind "DeploymentConfig" is registered for version "v1"
    not to have occurred
```

File: test/extended/deployments/util.go [hunk 2]
```diff
@@ -625,24 +614,17 @@ func readDCFixture(path string) (*appsapi.DeploymentConfig, error) {
...
+	appsCodecs := serializer.NewCodecFactory(appsScheme)
+	appsapiv1.AddToScheme(appsScheme)
+	obj, err := runtime.Decode(appsCodecs.UniversalDecoder(appsapiv1.SchemeGroupVersion), content)
```
Comment (mfojtik) [inline test/extended/deployments/util.go]: (this is where the error is comming from...)

File: test/extended/deployments/util.go (hunk 1 above)
Comment (mfojtik) [reply test/extended/deployments/util.go]: oh, the fixtures might be oapi...

Comment (mfojtik) [reply test/extended/deployments/util.go]: fixed. (converted all test fixtures to grouped api)

**Comment by tnozicka** [comment]
a nit - it would be nice to use `dcName` and `dc.Name` consistently. My rule of thumb in the other test was use`dcName` before the DC object is created, assert for equality and use`dc.Name` everywhere after that point because it makes it consistent with using other properties of that object.

**Comment by mfojtik** [comment]
@tnozicka i tried to use the dcName consistently, can you point me into a spot where we use dc.Name ?

**Comment by mfojtik** [comment]
@tnozicka all green

File: test/extended/testdata/deployments/multi-ict-deployment.yaml [hunk 3]
```diff
@@ -1,46 +1,43 @@
...
+  name: example
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: example
+    spec:
+      containers:
+      - imagePullPolicy: Always
```
[the remark this answers is not in the document]
Comment (mfojtik) [reply test/extended/testdata/deployments/multi-ict-deployment.yaml]: pre-existing, but I will change it

[the remark this answers is not in the document]
Comment (kargakis) [reply test/extended/testdata/deployments/multi-ict-deployment.yaml]: Can you also update this?

**Pull request merged**